core/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java

```diff
@@ -95,13 +92,4 @@ public Explanation explain(Query query, int doc) throws IOException {
             searchContext.clearReleasables(Lifetime.COLLECTION);
         }
     }
-
-    @Override
-    protected void search(List<LeafReaderContext> leaves, Weight weight, Collector collector) throws IOException {
-        try {
-            super.search(leaves, weight, collector);
-        } finally {
-            searchContext.clearReleasables(Lifetime.COLLECTION);
-        }
-    }
```

Comment:
can we remove the "clearReleasables" call in other methods like createWeight too?